src/components/ActivityCollector/createLinkClick.js

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+  getAbsoluteUrlFromAnchorElement,
+  isSupportedAnchorElement,
+  isDownloadLink,
+  isExitLink
+} from "./utils";
+
+const determineLinkType = (window, config, linkUrl, clickedObj) => {
+  let linkType = "other";
+  if (isDownloadLink(config.downloadLinkQualifier, linkUrl, clickedObj)) {
+    linkType = "download";
+  } else if (isExitLink(window, linkUrl)) {
+    linkType = "exit";
+  }
+  return linkType;
+};
+
+export default (window, config, clickedObject) => {
```

Comment:
I think this would be better encapsulated if you returned the lifecycle function instead of an object with two methods.

```
export default (window, config) => {
  return (event, clickedObject) => {
    ...
  };
};
```

Reply:
Agreed 👍